jakarta-eclipse/src/org/jakarta/lsp4e/JakartaLanguageClient.java

```diff
@@ -73,13 +75,38 @@ public boolean isCanceled() {
 			for (String uri : uris) {
 
 				URI u = JDTUtils.toURI(uri);
+
 				ICompilationUnit unit = JDTUtils.resolveCompilationUnit(u);
 				if (unit != null) {
 					// System.out.println("--class name: " + unit.getElementName());
 					IType[] alltypes;
+					IAnnotation[] allAnnotations;
 					try {
 						alltypes = unit.getAllTypes();
 						for (IType type : alltypes) {
+							allAnnotations = type.getAnnotations();
+							boolean isWebServletAnnotated = false;
+							boolean isHttpServletExtended = false;
+							for (IAnnotation annotation : allAnnotations) {
+								if (annotation.getElementName() == "WebServlet") {
+//									System.out.println("--Annotation name: " + annotation.getElementName());
+									isWebServletAnnotated = true;
+								}
+							}
+
+							String typeExtension = type.getSuperclassTypeSignature();
+//							System.out.println("--extension name: " + type.getSuperclassTypeSignature());
+							// "Q****;" is a signature style, indicating unresolved signature
+							if ((typeExtension != null) && typeExtension.compareTo("QHttpServlet;") == 0) {
+								isHttpServletExtended = true;
+							}
+
+							if (isWebServletAnnotated && !isHttpServletExtended) {
+								ISourceRange nameRange = JDTUtils.getNameRange(type);
+								Range range = JDTUtils.toRange(unit, nameRange.getOffset(), nameRange.getLength());
+								diagnostics.add(new Diagnostic(range, "Need to extend HttpServlet class after annotate WebServlet"));
+							}
+
 							IMethod[] methods = type.getMethods();
```

Comment:
Lets remove the sample diagnostic now

Reply:
Do you want me to remove the test diagnostic message?

I think the initial test diagnostic was removed by Raza already.